### PR TITLE
Feat/twin macro & styled components (#39)

### PR DESCRIPTION
**Pull request by Vroomfrondal**
* chore: twin.macro & SC boilerplate
- npm install twin macro, styled components, babel, TS-types
- vite config
- npm babel macro for styled components
- twin.d.ts and ts.config boilerplate

* refactor: BannerImage TW on <App />
* refactor: <Title /> TW & SC

* refactor: <ScrollToTopButton/> TW & SC
- refactored to styled Button with styled components/Twin macro
- increased readability

* refactor: <Loader /> TW & SC

* refactor: <CardModal /> TW & Styled Components
- added twin.macro to <CardModal />
- increased Modal size on mobile and desktop

* refactor: <Card /> TW & Styled Components
- refactored main class names

* feat: Error Page, Nav Bar TW & Styled Comp
- refactored <NavBar /> to use twin macro & styled components
- added <Error /> Page to account for when NASA Server is down (502 / 503 err)

* refactor: <Favorites /> Twin Macro & SC

### CODE REVIEW

**Comment by Vroomfrondal** [comment]
Apologies for the delay. Lots of work-work as well as the NASA API being down for a few days. Ended up being a good thing though - I hadn't added an error page to route users so I went ahead and mixed that in with this PR.

File: src/components/Pages/Error.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      </ErrorContainer>
+    </>
+  )
+}
+
+export default Error
+
+// Containers
+const ErrorContainer = tw.div`flex flex-col justify-center items-center h-[50rem] text-cream`
+const LinkContainer = tw.div`pt-10 flex flex-col justify-center items-center`
```
Comment (Idonoghue) [inline src/components/Pages/Error.tsx:36]: Another tool available in organizing styles is moving reused styles into a different component
```suggestion
// Containers
const Container = tw.div`flex flex-col justify-center items-center`
const ErrorContainer = tw(Container)`h-[50rem] text-cream`
const LinkContainer = tw(Container)`pt-10`
```

File: src/components/CardModal.tsx [hunk 2]
```diff
@@ -16,48 +17,70 @@ function CardModal({ image, isLikedImage, isOpen, onLike, onClose }: CardModalDa
...
+const Title = tw.span`absolute top-[8.5rem] bottom-0 left-0 right-0 w-full font-normal text-lg text-center flex justify-center items-center md:(text-xl)`
+
+const Author = tw.span`absolute top-1 left-1 text-xs opacity-70`
+
+const LikeButton = tw.span`absolute m-2 right-0 bottom-1 cursor-pointer font-[500] h-[1.3rem] w-[1.3rem] border-none hover:(opacity-75)`
+
+// Body Container Elements
+const DateEl = tw.span`text-lg text-left`
+
+const Salutations = tw.span`text-left pb-1 text-lg`
```
Comment (Idonoghue) [inline src/components/CardModal.tsx:84]: Here could be another spot for reuse
```suggestion
const PreText = tw.span`text-left text-lg`

const DateEl = tw(PreText)``

const Salutations = tw(PreText)`pb-1`
```

File: src/components/NavBar.tsx [hunk 3]
```diff
@@ -13,75 +14,81 @@ function NavBar({ setActivePage }: NavBarActions) {
...
-      >
+      <HamburgerContainer onClick={() => (isShowingLinks ? setIsShowingLinks(false) : null)}>
+        <Hamburger isShowingLinks={isShowingLinks} onClick={() => setIsShowingLinks((status) => !status)}>
+          <TempHamburgerIcon className="w-4" />
+          <TempHamburgerIcon className="w-5" />
+          <TempHamburgerIcon className="w-6" />
+        </Hamburger>
+      </HamburgerContainer>
+
+      <NavigationBar isShowingLinks={isShowingLinks}>
```
Comment (Idonoghue) [inline src/components/NavBar.tsx:25]: Makes this jsx soo much easier to read 🥳

**Review by Idonoghue — APPROVED** [review]
Excellent!! Just some a couple random pointers but lgtm

File: src/components/Pages/Error.tsx (hunk 1 above)
Comment (Vroomfrondal) [reply src/components/Pages/Error.tsx:36]: Super cool feature. Thanks!

File: src/components/NavBar.tsx (hunk 3 above)
Comment (Vroomfrondal) [reply src/components/NavBar.tsx:25]: I agree. I was really worried after the Tailwind Integration on how things looked but I like this way more after this integration!.

**Pull request merged**